All_materials/20230822_Web_Scraping_Financial_Ombudsman_Service/src/web_scraping_and_data_wrangling/fos_web_scrap.py

```diff
@@ -0,0 +1,335 @@
+            
+    return all_text
+
+def getting_numbers_from_string(string_with_numbers):
+    
+    numbers_only = [x for x in string_with_numbers if x.isdigit()]
+    numbers_only = ''.join(numbers_only)
+    
+    return numbers_only
+
+def finding_scale_of_money(string_text):
+    
+    #Need to seperation the stings
+    string_splint = string_text.split()
+    
+    money_mentions = [x for x in string_splint if '£' in x]
+    money_mentions = [getting_numbers_from_string(x) for x in money_mentions]
+    
+    multple_of_lost = [len(x) for x in money_mentions]
+    
+    if multple_of_lost:
+        
+        #Need to have a scale format
+        string_scale = '10^' +  str(max(multple_of_lost)-1)
+        return string_scale
+    
+    else:
+        
+        return '10^?'
+    
+def was_CRM_mentioned(string_text):
+    
+    tokenizer = RegexpTokenizer(r'\w+')
+    
+    list_of_strings = tokenizer.tokenize(string_text.lower())
+    
+    mentioned_CRM = [x for x in list_of_strings if 'crm' in x]
+    
+    if mentioned_CRM:
+        
+        return True
+    
+    else:
+        
+        return False
+    
+    
+def finding_scale_of_lost_in_complaint(ref_complaint):
+    
+    all_text = getting_pdf_text(ref_complaint)
+    
+    scale_of_lost = finding_scale_of_money(all_text)
+    
+    return scale_of_lost
+
+def is_CRM_used(binary_take):
+    
+    if binary_take:
+        return 'YES'
+    else:
+        return 'NO'
+    
+def lemmatize_stemming(text):
+    return stemmer.stem(WordNetLemmatizer().lemmatize(text, pos='v'))
+
+# Tokenize and lemmatize
+def preprocess(text):
+    result=[]
+    for token in gensim.utils.simple_preprocess(text) :
+        if token not in gensim.parsing.preprocessing.STOPWORDS and len(token) > 3:
+            result.append(lemmatize_stemming(token))
+            
+    return result
+
+def date_format_as_string_in_tweet(date_info):
+    
+    #getting the date
+    date_of_compl = date_info.strftime("%m%d%y")
+    
+    return date_of_compl
+
+def getting_a_single_tweet_from_complint(single_series):
+    
+    #Need to contruct the extract
+    full_text_complaint = single_series['all_text']
+    text_without_stopwords = word_tokenize(full_text_complaint)
+    text_without_stopwords = [word for word in text_without_stopwords if not word in all_stopwords]
+    text_without_stopwords = (" ").join(text_without_stopwords)
+    
+    #Need to get a 1/4 of the point
+    index_start = round(len(text_without_stopwords)/4)
+    text_without_stopwords = text_without_stopwords[index_start:]
+    
+    string_consturction = (
+        date_format_as_string_in_tweet(single_series['date']) + '\n' +
+        single_series['ref'] + '\n' + 
+        '~£' + single_series['scale_of_lost'] + '\n' +
+        single_series['outcome'] + '\n' +
+        'CRM?' + is_CRM_used(single_series['CRM']) + '\n' +
+        '...' + text_without_stopwords
+        )
+    
+    #Crop
+    string_consturction = string_consturction[:279]
```

Comment:
```suggestion
    string_construction = string_construction[:279]
```
correction needed throughout

Reply:
This was based on twitter's character limit - A nugget size information

Reply:
This is try get it in a tweet format